index.bs

```diff
@@ -12995,207 +12875,291 @@ with the following characteristics:
-*   <div algorithm="to invoke the set method of Maps">
+then there must exist a <code class="idl">set</code> data property
+on |A|’s [=interface prototype object=]
+with the following characteristics:
 
-        The value of the property is a [=built-in function object=] that behaves as follows when invoked:
+*   The property has attributes { \[[Writable]]: <emu-val>true</emu-val>, \[[Enumerable]]: <emu-val>false</emu-val>, \[[Configurable]]: <emu-val>true</emu-val> }.
+*   The value of the property is a [=built-in function object=]
+    that behaves as follows when invoked:
 
-        1.  Let |O| be the <emu-val>this</emu-val> value.
-        1.  If |O| [=is a platform object=],
-            then [=perform a security check=], passing:
-            *   the platform object |O|,
-            *   the identifier "<code>set</code>", and
-            *   the type "<code>method</code>".
-        1.  If |O| does not [=implement=] <var ignore>A</var>, then [=ECMAScript/throw=] a {{ECMAScript/TypeError}}.
-        1.  Let |map| be the {{ECMAScript/Map}} object that is the value of |O|’s \[[BackingMap]] [=internal slot=].
-        1.  Let |keyType| and |valueType| be the key and value types specified in the [=maplike declaration=].
-        1.  Let |function| be [=!=] <a abstract-op>Get</a>(|map|, "<code>set</code>").
+    <div algorithm="to invoke the set method of Maps">
+        1.  Let |O| be the <emu-val>this</emu-val> value, [=implementation-checked=] against <var ignore>a</var> with identifier "<code>set</code>" and type "<code>method</code>".
+        1.  Let |map| be |O|'s [=map entries=].
+        1.  Let |keyType| be the key type specified in the [=maplike declaration=], and |valueType| be the value type.
         1.  Let |keyArg| be the first argument passed to this function, or <emu-val>undefined</emu-val> if not supplied.
+        1.  Let |key| be |keyArg| [=converted to an IDL value=] of type |keyType|.
+        1.  If |key| is -0, set |key| to +0.
         1.  Let |valueArg| be the second argument passed to this function, or <emu-val>undefined</emu-val> if not supplied.
-        1.  Let |keyIDL| be the result of [=converted to an IDL value|converting=] |keyArg| to an IDL value of type |keyType|.
-        1.  Let |valueIDL| be the result of [=converted to an IDL value|converting=] |valueArg| to an IDL value of type |valueType|.
-        1.  Let |key| be the result of [=converted to ECMAScript values|converting=] |keyIDL| to an ECMAScript value.
-        1.  Let |value| be the result of [=converted to ECMAScript values|converting=] |valueIDL| to an ECMAScript value.
-        1.  Perform [=?=] <a abstract-op>Call</a>(|function|, |map|, «|key|, |value|»).
-        1.  Return |O|.
+        1.  Let |value| be |valueArg| [=converted to an IDL value=] of type |valueType|.
+        1.  [=map/Set=] |map|[|key|] to |value|.
+        1.  Return |value| [=converted to an ECMAScript value=].
```

Comment:
This looks like an unintentional normative change. You need to return O.

Reply:
Whoops, indeed, not sure why I thought it returned the value. I know I wrote that on purpose, huh.